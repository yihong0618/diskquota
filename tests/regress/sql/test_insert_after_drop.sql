CREATE DATABASE db_insert_after_drop;
\c db_insert_after_drop
CREATE EXTENSION diskquota;
-- Test Drop Extension
CREATE SCHEMA sdrtbl;
SELECT diskquota.set_schema_quota('sdrtbl', '1 MB');
SET search_path TO sdrtbl;
CREATE TABLE a(i int) DISTRIBUTED BY (i);
INSERT INTO a SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,100000);
SELECT diskquota.wait_for_worker_new_epoch();
INSERT INTO a SELECT generate_series(1,100);
DROP EXTENSION diskquota;
INSERT INTO a SELECT generate_series(1,100);

DROP TABLE a;
\c postgres
DROP DATABASE db_insert_after_drop;
