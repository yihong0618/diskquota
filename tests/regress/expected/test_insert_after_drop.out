CREATE DATABASE db_insert_after_drop;
\c db_insert_after_drop
CREATE EXTENSION diskquota;
-- Test Drop Extension
CREATE SCHEMA sdrtbl;
SELECT diskquota.set_schema_quota('sdrtbl', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO sdrtbl;
CREATE TABLE a(i int) DISTRIBUTED BY (i);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO a SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO a SELECT generate_series(1,100000);
SELECT pg_sleep(10);
 pg_sleep 
----------
 
(1 row)

INSERT INTO a SELECT generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name: sdrtbl
DROP EXTENSION diskquota;
-- sleep 1 second in case of system slow
SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

INSERT INTO a SELECT generate_series(1,100);
DROP TABLE a;
\c postgres
DROP DATABASE db_insert_after_drop;
